File: leetcode/AIcoding/2.Gemini-2.0/javascript_solutions/1899-merge-triplets-to-form-target-triplet.js
var mergeTriplets = function(triplets, target) {
    let good = [0, 0, 0];
    for (let triplet of triplets) {
        if (triplet[0] <= target[0] && triplet[1] <= target[1] && triplet[2] <= target[2]) {
            if (triplet[0] === target[0]) good[0] = 1;
            if (triplet[1] === target[1]) good[1] = 1;
            if (triplet[2] === target[2]) good[2] = 1;
        }
    }
    return good[0] + good[1] + good[2] === 3;
};

File: leetcode/AIcoding/2.Gemini-2.0/rust_solutions/1899-merge-triplets-to-form-target-triplet.rs
impl Solution {
    pub fn merge_triplets(triplets: Vec<Vec<i32>>, target: Vec<i32>) -> bool {
        let mut good1 = false;
        let mut good2 = false;
        let mut good3 = false;

        for triplet in triplets {
            if triplet[0] <= target[0] && triplet[1] <= target[1] && triplet[2] <= target[2] {
                if triplet[0] == target[0] && triplet[1] <= target[1] && triplet[2] <= target[2] {
                    good1 = true;
                }
                if triplet[0] <= target[0] && triplet[1] == target[1] && triplet[2] <= target[2] {
                    good2 = true;
                }
                if triplet[0] <= target[0] && triplet[1] <= target[1] && triplet[2] == target[2] {
                    good3 = true;
                }
            }
        }

        good1 && good2 && good3
    }
}